README.md

```diff
@@ -1,2 +1,283 @@
+
+# Table of Contents
+1. [Quick start](#quick-start)
+2. [API Reference](#api-reference)
+3. [Initialization and usage](#initialization-and-usage)
+4. [Advanced usage](#advanced-usage)
+
+
+## Quick start
+
+### Install as Gradle dependency
+
+Add these lines to `build.gradle` file of your project.
+
+
+```gradle
+allprojects {
+	repositories {
+		...
+		maven { url 'https://jitpack.io' }	
+	}
+}
+```
+
+### Add the dependency to your project
+
+Add these lines to `build.gradle` file of your module.
+This library depends on `kotlin-stdlib`. If your application is written in Java, add `kotlin-stdlib` dependency first (it's lightweight and has excellent backward and forward compatibility).
+
+```gradle
+dependencies {
+	implementation 'com.github.fingerprintjs:fingerprint-android:1.0.0'
+	
+	// Add this line for java-project
+	implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"	
+}
+```
+
+## API Reference
+
+### Device ID and Fingerprint
+
+The library operates with two entities. 
+
+1. `deviceId` - it is an ID of the device, based on system-provided IDs. Currently, it is based on Google Service Framework ID, and ANDROID_ID if the first is unavailable. It's strict, which means the uniqueness of the ID for every device. Also, it keeps the same after application reinstalling.
+2. `fingerprint` - it is a digital fingerprint of a device. Basically, it is a hash, calculated from a lot of available platform signals. `fingerprint` **is not a strict ID**, so there is a probability that two different devices will have the same fingerprint. Also, there is a probability, that the same device will have a different fingerprint in different moments due to system update, settings changing, etc.
+
+#### Why do we need them both?
+It is a good question, most of use-cases can be covered with the `deviceId`. But the `GSF_ID` and `ANDROID_ID`can be spoofed much easier, than device data such as CPU info or sensors list. 
+
+Another reason for using `fingerprint` is a possible restriction of access to `deviceId` due to privacy policy changes trend.
+
+
+
+### Initialization and usage
+
+Initialize the library and get `fingerprint` and `deviceId`.
+
+In Kotlin
+
+```kotlin
+
+// Initialization
+val fingerprinter =
+            FingerprinterFactory.getInitializedInstance(applicationContext)
+
+// Simple usage
+val fingerprint = fingerprinter.fingerprint()
+val deviceId = fingerprinter.deviceId()
+
+```
+
+In Java
+
+```java
+
+// Initialization
+Fingerprinter fingerprinter =
+                FingerprinterFactory.getInitializedInstance(getApplicationContext());
+                
+// Simple usage
+String fingerprint = fingerprinter.fingerprint();
+String deviceId = fingerprinter.deviceId();
+
+```
+### Advanced usage
+The additional entities, provided by the library are `BaseFingerprinter` and `Configuration`.
+
+#### BaseFingerprinter
+
+`fingerprint` can be calculated using different platform signals. They are combined in several groups:
```

Comment:
You say that signals are combined into groups. But then you call those groups "fingerprints", which is confusing. 
I would change to:

A device `fingerprint` can be calculated using various platform signals. 
These signals are grouped into several categories:

1. Hardware signals...
2. OS build signals & attributes ..
3. Device state information ..
4. Installed apps information ..

Reply:
then explain that every group of signals has a signal provider, which can be accessed for raw data. 
the default fingerprinter uses all the signal providers in the best way possible to generate a stable and unique identifier.